contracts/strategies/NFTVoteWinnerTakeAll.sol

```diff
@@ -67,17 +65,19 @@ contract NFTVoteWinnerTakeAll is BaseStrategy {
         payable
         onlyDuringAllocationPeriod
         onlyAllo
+        onlyPoolManager(_sender)
         returns (address)
     {
         address[] memory newRecipients = abi.decode(_data, (address[]));
 
-        for (uint256 i; i < newRecipients.length;) {
+        uint256 numRecipientsLength = newRecipients.length;
+        for (uint256 i; i < numRecipientsLength;) {
             isRecipient[newRecipients[i]] = true;
-            recipients.push(newRecipients[i]);
             unchecked {
                 ++i;
             }
         }
+        return address(0);
```

Comment:
This function at the interface level feels wrong.

`registerRecipients` -> if it registers -> returning 1 address -> doesn't make sense 

would it be an overkill to have registerRecipients meant for pool manager and register meant for user